x/oracle/keeper/aggregator/worker.go

```diff
@@ -20,14 +20,74 @@ type worker struct {
 	// when enough data(exceeds threshold) collected, aggregate to conduct the final price
 	a   *aggregator
 	ctx *AggregatorContext
+	// TODO: move outside into context through .ctx
+	records recordMsg
 }
 
-func (w *worker) do(msg *types.MsgCreatePrice) []*types.PriceSource {
-	accAddress, _ := sdk.AccAddressFromBech32(msg.Creator)
+// recordKey used to retrieve messages from records to evaluate that if a validator report proper price for a specific feederID+sourceID
+type recordKey struct {
+	validator string
+	feederID  uint64
+	sourceID  uint64
+}
+
+// recordMsg define wrap the map for fast access to validator's message info
+type recordMsg map[recordKey][]*types.PriceTimeDetID
+
+func newRecordMsg() recordMsg {
+	return make(map[recordKey][]*types.PriceTimeDetID)
+}
+
+func (r recordMsg) get(validator string, feederID, sourceID uint64) []*types.PriceTimeDetID {
+	v := r[recordKey{validator, feederID, sourceID}]
+	return v
+}
+
+func (r recordMsg) check(validator string, feederID, sourceID uint64, price, detID string) (exist, matched bool) {
+	prices := r.get(validator, feederID, sourceID)
+	for _, p := range prices {
+		if p.DetID == detID {
+			exist = true
+			if p.Price == price {
+				matched = true
```

Comment:
Use early return in the this  method to optimize performance like 

>   if p.DetID == detID {
            exist = true
           if p.Price == price {
               return true, true
           }
            return true, false
        }